include/nlohmann/json.hpp

```diff
@@ -3978,10 +3893,21 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     @brief overload for a default value of type const char*
     @copydoc basic_json::value(const typename object_t::key_type&, const ValueType&) const
     */
-    string_t value(const typename object_t::key_type& key, const char* default_value) const
+#if defined(JSON_HAS_CPP_17) // avoid creating a string_t value from default_value
+    template < class KeyType, typename detail::enable_if_t <
+                   detail::is_usable_as_key_type<basic_json_t, KeyType>::value, int> = 0>
```

Comment:
Minor, since this `template` clause is the same for `#if` and `#else` you could move it outside and avoid duplicating it.  If you prefer it this way, that's fine too.